src/mscorlib/src/System/StubHelpers.cs

```diff
@@ -27,7 +27,8 @@ internal static class AnsiCharMarshaler
         unsafe static internal byte[] DoAnsiConversion(string str, bool fBestFit, bool fThrowOnUnmappableChar, out int cbLength)
         {
             byte[] buffer = new byte[(str.Length + 1) * Marshal.SystemMaxDBCSCharSize];
-            fixed (byte *bufferPtr = buffer)
+            BCLDebug.Assert(buffer.Length != 0);
```

Comment:
Are you asking why the assert or why `BCLDebug.Assert`? I guess the assert isn't strictly necessary here -- I decided to add it in case `Marshal.SystemMaxDBCSCharSize` ever returned 0 (the way it is currently implemented it will always be greater than 0). I used `BCLDebug.Assert` because it was used elsewhere in this file.